src/getuuid.c

```diff
@@ -1,27 +1,71 @@
+#include <assert.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
 #include <ctype.h>
 
 #include "config.h"
 
+#include "utils/strutils.h"
 #include "utils/uuid.h"
 #include "utils/uuid4.h"
 #include "getuuid.h"
 
 
+/* Returns non-zero if `s` is a valid UUID. */
+
+int isuuid(const char *s)
+{
+  int i;
+  for (i=0; i<8; i++) if (!isxdigit(*(s++))) return 0;
+  if (*(s++) != '-') return 0;
+  for (i=0; i<4; i++) if (!isxdigit(*(s++))) return 0;
+  if (*(s++) != '-') return 0;
+  for (i=0; i<4; i++) if (!isxdigit(*(s++))) return 0;
+  if (*(s++) != '-') return 0;
+  for (i=0; i<4; i++) if (!isxdigit(*(s++))) return 0;
+  if (*(s++) != '-') return 0;
+  for (i=0; i<12; i++) if (!isxdigit(*(s++))) return 0;
+  return 1;
+}
+
+/* Returns non-zero if `s` matches <URI>/<UUID>. `len` is the length of `s`.
+   An optional final hash or slash will be ignored.*/
+int isinstanceuri(const char *s, int len)
+{
+  if (!len) len = strlen(s);
```

Comment:
If I read this correctly, you sort of make the length of s optional? Does that make sense? Would it not be better to either raise an error, or remove the argument and always calculate the length?

Reply:
You read it correctly. You are right, it doesn't make much sense (except for simplifying the tests) since this is an internal help function which probably should have been declared static. The only reason that it is not static is because that would make it inaccessible for unit testing...